utils/functions.sh

```diff
@@ -139,17 +139,36 @@ notify_error()
   notify "$_MESSAGE"
 }
 
+# Generates a backup of the current DB
+backup_db()
+{
+  message_step "Generating the backup DB dump:"
+  # Store a security backup in case the update doesn't go right.
+  DUMP_NAME="update-backup-script-$(date +%Y%m%d%H%M%S).sql";
+  DUMP_PATH="$WEBROOT/../dumps/$DUMP_NAME"
+  mkdir -p "$WEBROOT/../dumps/"
+  message_action "Dump generation in progress..."
+  $DRUSH sql-dump --result-file=$DUMP_PATH --gzip
+  # Remove older backups but keep the 10 youngest ones.
+  if [ "$(ls -l $WEBROOT/../dumps/*.sql.gz | wc -l)" -gt 10 ]; then
+    message_action "Cleaning up oldest backup dumps..."
+    ls -tp $WEBROOT/../dumps/*.sql.gz | grep -v '/$' | tail -n +10 | tr '\n' '\0' | xargs -0 rm --
+    message_confirm "Cleanup done!"
+  fi
+  message_confirm "Backup dump generated!"
+}
+
 # Download dump function.
 download_dump()
 {
   message_step "Updating the reference dump:"
   if [ -z "$COMBAWA_DB_FETCH_CNX_STRING" ]; then
     message_action "Copying locally the dump file..."
-    cp $COMBAWA_DB_FETCH_PATH "$COMBAWA_ROOT/$COMBAWA_DUMP_FILE_NAME.gz"
+    $($COMBAWA_DB_FETCH_COMMAND);
     message_confirm "Done!"
   else
     message_action "Fetching the dump from remote source..."
-    scp $COMBAWA_DB_FETCH_CNX_STRING:$COMBAWA_DB_FETCH_PATH "$COMBAWA_ROOT/$COMBAWA_DUMP_FILE_NAME.gz"
+    $($COMBAWA_DB_FETCH_COMMAND);
     message_confirm "Done!"
   fi
   if [[ $? != 0 ]]; then
```

Comment:
This needs more change. The `COMBAWA_DB_FETCH_CNX_STRING` variable is only written in the `.combawa` file for now so it's not accurate (and it looks like it shouldn't be there either).